lm-sensors: Add support for dev_name formatted like "a000000.wifi"

7217043955 opened the pull request:
Maintainer: @jow- Jo-Philipp Wich <jow@openwrt.org>
Compile tested: ipq4019, a customized router, v21.02.3 and v22.03.0
Run tested: ipq4019, a customized router, v21.02.3 and v22.03.0, see below mentioned issue 

Description:
Detail at #10623 
I found out that openwrt can patch patches on its own, lm-sensors repository seems not very active now, so we can fix it here now until upstream fix it.

7217043955 commented:
@hnyman

7217043955 commented:
Oops, the checks failed with message `Dirty patches detected, please refresh and review the diff `, but when I implement this change and run `make package/lm-sensors/{clean,prepare} V=sc`, it says 
```
Applying ./patches/001-Add_support_for_dev_name_formatted_like_a000000.wifi.patch using plaintext:
patching file lib/sysfs.c
```
and patchs without a problem, shall I recreate the patch using the openwrt package source instead of upstream master?

Sorry for the inconvenient, can I just say:
Signed-off-by: @7217043955 ?

hnyman commented:
> when I implement this change and run make package/lm-sensors/{clean,prepare} V=sc, it says

Do refresh instead:
`make package/lm-sensors/{clean,refresh} V=sc`

(It might be something small, like one char length difference in patch context header line.)

> can I just say:
> Signed-off-by: @7217043955 ?

No. Similar as Linux upstream, you need a proper name and mail address.  (instead of noreply-github...)
https://openwrt.org/submitting-patches#sign_your_work

7217043955 commented:
Ran the `make package/lm-sensors/{clean,refresh} V=sc`command, and updated the patch file to the one it generated, still can't pass the check, I even tried to using the packages master branch instead of openwrt v22.03.0 feed.

7217043955 commented:
[https://github.com/openwrt/packages/pull/19357](url)
This one also failed because of dirty patches, but they only changed a bunch of Makefile.

7217043955 commented:
Is Signed-off-by and lm-sensors reference ok now?

hnyman commented:
> Is Signed-off-by and lm-sensors reference ok now?

 Likely Aquamarine is not your real name, so no. 

The lm-sensors part is right.

7217043955 commented:
@1715173329 Thank you but I still don't know how to solve the dirty patches issue, any idea on that?

1715173329 commented:
> @1715173329 Thank you but I still don't know how to solve the dirty patches issue, any idea on that?

Need dos2unix

7217043955 commented:
@1715173329 Thank you again! It worked! I never pay attention to it, I was doing it on a Linux machine it's strange it created a CRLF file.
Now I squashed the commits and changed the signed-off, see if that's ok now.

1715173329 commented:
Nah, you need update git config.

```bash
git config user.name 'Florence Riker'
git config user.email '1460830@gmail.com'
git commit --amend --reset-author --no-edit
git push -f
```

Thanks.
